### ISSUE TITLE
Project Stage

### ISSUE THREAD

**Issue by btd1337** [Lomeli12/Beaver#6]
Does this project already work?

**Comment by Lomeli12** [comment]
It can be compiled and used to build simple Vala projects. You can even build beaver with beaver. It just doesn't have all the features I want in yet, and I haven't been able to work on it as my laptop broke (though that should change as a new one is coming in tomorrow).

**Comment by btd1337** [comment]
```
Compilation failed: 1 error(s), 39 warning(s)
 - /usr/bin/ld: /tmp/cciRLyjo.o: na função "beaver_beaver_main":
 - Beaver.vala.c:(.text+0x282): referência não definida para "rintf"
 - collect2: error: ld returned 1 exit status
 - error: cc exited with status 256
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/CommandContext.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/arguments/Argument.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/arguments/BoolArgument.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/arguments/DoubleArgument.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/arguments/FloatArgument.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/arguments/IntArgument.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/arguments/LongArgument.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/lib/cmd/arguments/StringArgument.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/logging/ConFormat.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/logging/Logger.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/parse/Parser.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/project/BeaverInfo.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/project/BeaverProject.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - Compilation failed: 1 error(s), 39 warning(s)
 - cc1: fatal error: /mnt/Dados/Repositories/Beaver/src/main/vala/project/BuildEnvironment.vala.c: Arquivo ou diretório inexistente
 - compilation terminated.
 - error: cc exited with status 256
```

**Comment by Lomeli12** [comment]
Do you have all the dependencies installed and are using the `valac` build command from the README? Also what are you building it on? My current setup I can build it on Windows via MSYS2, Linux Mint 19.1, and Manjaro 18.0 with no problems.

**Comment by btd1337** [comment]
I'm trying build beaver...

Because when I run the command below an error occur.

```
beaver build
zsh: command not found: beaver
```

I'm running on Arch Linux.

**Comment by Lomeli12** [comment]
You need to build beaver at least once via `valac`. Clone the repo, cd into the repo, then run the `valac` command in the README to build it.

```
git clone https://github.com/Lomeli12/Beaver
cd Beaver
valac -o beaver --pkg json-glib-1.0 --pkg gio-2.0 --pkg posix src/main/vala/Beaver.vala src/main/vala/lib/*.vala src/main/vala/lib/cmd/*.vala src/main/vala/lib/cmd/arguments/*.vala src/main/vala/logging/*.vala src/main/vala/project/*.vala src/main/vala/parse/*.vala
```

Then you can try beaver using:

```
.\beaver
```

**Comment by btd1337** [comment]
```
$ valac -o beaver --pkg json-glib-1.0 --pkg gio-2.0 --pkg posix src/main/vala/Beaver.vala src/main/vala/lib/*.vala src/main/vala/lib/cmd/*.vala src/main/vala/lib/cmd/arguments/*.vala src/main/vala/logging/*.vala src/main/vala/project/*.vala src/main/vala/parse/*.vala


src/main/vala/logging/ConFormat.vala:4.9-4.37: warning: the modifier `static' is not applicable to constants
        public static const int BLACK = 30;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:5.9-5.35: warning: the modifier `static' is not applicable to constants
        public static const int RED = 31;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:6.9-6.37: warning: the modifier `static' is not applicable to constants
        public static const int GREEN = 32;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:7.9-7.38: warning: the modifier `static' is not applicable to constants
        public static const int YELLOW = 33;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:8.9-8.36: warning: the modifier `static' is not applicable to constants
        public static const int BLUE = 34;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:9.9-9.39: warning: the modifier `static' is not applicable to constants
        public static const int MAGENTA = 35;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:10.9-10.36: warning: the modifier `static' is not applicable to constants
        public static const int CYAN = 36;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:11.9-11.37: warning: the modifier `static' is not applicable to constants
        public static const int WHITE = 37;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:14.10-14.41: warning: the modifier `static' is not applicable to constants
         public static const int BLACK_BG = 40;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:15.10-15.39: warning: the modifier `static' is not applicable to constants
         public static const int RED_BG = 41;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:16.10-16.41: warning: the modifier `static' is not applicable to constants
         public static const int GREEN_BG = 42;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:17.10-17.42: warning: the modifier `static' is not applicable to constants
         public static const int YELLOW_BG = 43;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:18.10-18.40: warning: the modifier `static' is not applicable to constants
         public static const int BLUE_BG = 44;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:19.10-19.43: warning: the modifier `static' is not applicable to constants
         public static const int MAGENTA_BG = 45;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:20.10-20.40: warning: the modifier `static' is not applicable to constants
         public static const int CYAN_BG = 46;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:21.10-21.41: warning: the modifier `static' is not applicable to constants
         public static const int WHITE_BG = 47;
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:24.9-24.37: warning: the modifier `static' is not applicable to constants
        public static const int RESET = 0;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:25.9-25.36: warning: the modifier `static' is not applicable to constants
        public static const int BOLD = 1;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:26.9-26.38: warning: the modifier `static' is not applicable to constants
        public static const int BRIGHT = 1;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:27.9-27.41: warning: the modifier `static' is not applicable to constants
        public static const int UNDERLINE = 4;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:28.9-28.39: warning: the modifier `static' is not applicable to constants
        public static const int INVERSE = 7;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:29.9-29.40: warning: the modifier `static' is not applicable to constants
        public static const int BOLD_OFF = 21;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:30.9-30.42: warning: the modifier `static' is not applicable to constants
        public static const int BRIGHT_OFF = 21;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:31.9-31.45: warning: the modifier `static' is not applicable to constants
        public static const int UNDERLINE_OFF = 24;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/logging/ConFormat.vala:32.9-32.43: warning: the modifier `static' is not applicable to constants
        public static const int INVERSE_OFF = 27;            
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/project/BuildEnvironment.vala:6.9-6.48: warning: the modifier `static' is not applicable to constants
        public static const string BEAVER_FOLDER = ".beaver";
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/project/BuildEnvironment.vala:7.9-7.45: warning: the modifier `static' is not applicable to constants
        public static const string BUILD_FILE = "build.beaver";
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/project/BuildEnvironment.vala:8.9-8.47: warning: the modifier `static' is not applicable to constants
        public static const string BUILD_FOLDER = "build/";
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/project/BuildEnvironment.vala:9.9-9.48: warning: the modifier `static' is not applicable to constants
        public static const string SOURCE_FOLDER = "src/";
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/project/BuildEnvironment.vala:10.9-10.46: warning: the modifier `static' is not applicable to constants
        public static const string MAIN_FOLDER = SOURCE_FOLDER + "main/";
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/project/BuildEnvironment.vala:11.9-11.46: warning: the modifier `static' is not applicable to constants
        public static const string CODE_FOLDER = MAIN_FOLDER + "vala/";
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/project/BuildEnvironment.vala:12.9-12.46: warning: the modifier `static' is not applicable to constants
        public static const string VAPI_FOLDER = MAIN_FOLDER + "vapi/";
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
src/main/vala/parse/Parser.vala:11.17-11.27: warning: `null' incompatible with return type `Beaver.Project.BeaverProject'
                return null;
                ^^^^^^^^^^^
src/main/vala/parse/Parser.vala:16.17-16.27: warning: `null' incompatible with return type `Beaver.Project.BeaverProject'
                return null;
                ^^^^^^^^^^^
src/main/vala/parse/Parser.vala:47.17-47.27: warning: `null' incompatible with return type `Beaver.Project.BeaverInfo'
                return null;
                ^^^^^^^^^^^
src/main/vala/parse/Parser.vala:54.17-54.27: warning: `null' incompatible with return type `Beaver.Project.BeaverInfo'
                return null;
                ^^^^^^^^^^^
src/main/vala/parse/Parser.vala:59.17-59.27: warning: `null' incompatible with return type `Beaver.Project.BeaverInfo'
                return null;
                ^^^^^^^^^^^
src/main/vala/parse/Parser.vala:30.21-30.31: warning: `null' incompatible with return type `Beaver.Project.BeaverProject'
                    return null;
                    ^^^^^^^^^^^
src/main/vala/parse/Parser.vala:41.13-41.23: warning: `null' incompatible with return type `Beaver.Project.BeaverProject'
            return null;
            ^^^^^^^^^^^
In file included from /usr/include/glib-2.0/glib.h:82,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /home/arch/Repositories/Beaver/src/main/vala/lib/StringUtil.vala.c:4:
/home/arch/Repositories/Beaver/src/main/vala/lib/StringUtil.vala.c: In function ‘string_strip’:
/home/arch/Repositories/Beaver/src/main/vala/lib/StringUtil.vala.c:73:14: warning: passing argument 1 of ‘g_strchug’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   73 |  g_strstrip (_tmp1_);
      |              ^~~~~~
/usr/include/glib-2.0/glib/gstrfuncs.h:181:53: note: in definition of macro ‘g_strstrip’
  181 | #define g_strstrip( string ) g_strchomp (g_strchug (string))
      |                                                     ^~~~~~
/usr/include/glib-2.0/glib/gstrfuncs.h:176:23: note: expected ‘gchar *’ {aka ‘char *’} but argument is of type ‘const gchar *’ {aka ‘const char *’}
  176 | gchar*                g_strchug        (gchar        *string);
      |                       ^~~~~~~~~
/usr/bin/ld: /tmp/ccwOfDBo.o: in function "beaver_beaver_main":
Beaver.vala.c:(.text+0x282): reference not set to "rintf"
collect2: error: ld returned 1 exit status
error: cc exited with status 256
Compilation failed: 1 error(s), 39 warning(s)
```

## OS Informations:

```
Package: Version
vala: 0.44.7-1
json-glib (json-glib-1.0): 1.4.4-1
glib2 (gio-2.0): 2.62.0-1
glibc (posix): 2.29-4
```

**Comment by Lomeli12** [comment]
I made a fresh Arch VM and was able to replicate the error. I also realized I forgot to link the Math library, which is causing the issue. Simple fix though.
